src/main/java/school/lemon/changerequest/java/extendedinteger/ExtendedInteger.java

```diff
@@ -1,119 +1,242 @@
 package school.lemon.changerequest.java.extendedinteger;
 
+import java.util.Random;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /**
  * Integer representation with some additional features
  */
 public class ExtendedInteger {
 
+    int intValue;
+
     public ExtendedInteger(int value) {
-        //TODO: implement me
+        intValue = value;
     }
 
     /**
      * Check whether specified value is even
+     *
      * @param value to check
      * @return true if value is even, false - otherwise
      */
     public static boolean isEven(int value) {
-        //TODO: implement me
-        return false;
+
+        boolean Even = ((value & 1) == 0);
+        return Even;
     }
 
     /**
      * Check whether specified value is odd
+     *
      * @param value to check
      * @return true if value is odd, false - otherwise
      */
     public static boolean isOdd(int value) {
-        //TODO: implement me
+        boolean Odd = ((value & 1) == 1);
+        return Odd;
+    }
+
+    /**
+     * Binary method of exponentiation
+     *
+     * @param M   basis of degree
+     * @param Key exponent
+     * @param N   module
+     * @param C   result
+     */
+    private static void pow_mod(int M, int Key, int N, int[] C) {
```

Comment:
Really? Are you kidding me?

1) method name should be in `camelCase`.
2) variable names should be in `camelCase` and have meaningful names.
3) why do you need this method for? It's hardly maintainable and probably useless.